### PR TITLE
Fix Language Switcher in single adventure

### PR DESCRIPTION
**Pull request by babaMar**
**Description**

_The request handler for single adventure was not handling the `keyword_language` parameter. In turn the language to be passed to `load_adventures_per_level` function was always the language in app context `g`. At the same time, in `load_adventures_per_level` function the language argument was not to use as key for `ADVENTURES`.

Fixes for code style._

**Fixes https://github.com/Felienne/hedy/issues/3485 **

It probably has some impact on https://github.com/Felienne/hedy/issues/3433 as well. In the top-right corner I don't see the language switcher though, only my username, so not able to test that.

**How to test**
Run locally, and browse to single adventure. Use the translation button to switch language

### CODE REVIEW

**Comment by TiBiBa** [comment]
Hi @babaMar! Thanks for your contribution to Hedy! In your PR it is hard to find the code changes you made because a lot of (automatic?) is performed on the related files. This makes it hard for me to review. Could you reverse the styling changes and only commit the code changes? Thanks!

**Comment by babaMar** [comment]
> Hi @babaMar! Thanks for your contribution to Hedy! In your PR it is hard to find the code changes you made because a lot of (automatic?) is performed on the related files. This makes it hard for me to review. Could you reverse the styling changes and only commit the code changes? Thanks!

Hey, you're right, that's what I also thought. I installed the pre-commit hook, so it doesn't let me commit only those changes :/

**Comment by TiBiBa** [comment]
> > Hi @babaMar! Thanks for your contribution to Hedy! In your PR it is hard to find the code changes you made because a lot of (automatic?) is performed on the related files. This makes it hard for me to review. Could you reverse the styling changes and only commit the code changes? Thanks!
> 
> Hey @TiBiBa, you're right, that's what I also thought. I installed the pre-commit hook, so it doesn't let me commit only those changes :/
> 
> The main changes are (this solves the problem of https://hedy.org/adventure/parrot/1?keyword_language=nl not doing anything):
> 
> ```python
> @app.route('/adventure/<name>', methods=['GET'], defaults={'level': 1})
> @app.route('/adventure/<name>/<level>', methods=['GET'])
> def get_specific_adventure(name, level):
>     app_logger.info(f'Requested adventure {name} with language {g.keyword_lang} '
>                     f'and language: {request.args.get("keyword_language")}')
>     language = request.args.get("keyword_language", g.keyword_lang)
> 
>     # ... and then passing it to subsequent calls of
>     adventure = [x for x in load_adventures_per_level(
>         level, language)
>     # ...   
>     
> ```
> 
> And then in (this is very heuristic so probably I need to better understand the system):
> 
> ```python
> def load_adventures_per_level(level, keyword_lang):
>     app_logger.debug(f"loading adventure {level=} for language {keyword_lang}")
>     # .....
>     all_adventures = []
>     # I use the argument keyword_lang as a key for the ADVENTURES collection 
>     adventures = ADVENTURES[keyword_lang].get_adventures(keyword_lang)
>     # ....
> ```
> 
> Also as a side note: In the UI top-right corner I don't see the language switcher though, only my username, so not able to test that.

I see! No problem, I will just dive into the code and figure it out, no problem!

**Review by TiBiBa — CHANGES_REQUESTED** [review]
It is very hard for me to review this PR as there are many changes going on. Please refrain from making changes unrelated to the related issue when submitting a PR. It is still unclear to me what the actual fix is for the related issue.

File: hedy_content.py [hunk 1]
```diff
@@ -89,10 +95,10 @@
...
         languages[folder] = locale.display_name.title()
 
-for l in sorted(languages):
-    ALL_LANGUAGES[l] = languages[l]
-    if os.path.exists('./grammars/keywords-' + l + '.lark'):
-        ALL_KEYWORD_LANGUAGES[l] = l[0:2].upper()  # first two characters
+for _lang in sorted(languages):
+    ALL_LANGUAGES[_lang] = languages[_lang]
+    if os.path.exists('./grammars/keywords-' + _lang + '.lark'):
+        ALL_KEYWORD_LANGUAGES[_lang] = _lang[0:2].upper()  # first two characters
```
[the remark this answers is not in the document]
Comment (babaMar) [reply hedy_content.py:101]: The pre-commit hook complained with "ambiguous variable name"

File: hedy_content.py [hunk 2]
```diff
@@ -135,11 +141,11 @@ def cache_keyword_parsing(self, language):
                     try:
                         command[k] = v.format(**KEYWORDS.get(language))
                     except IndexError:
-                        print("There is an issue due to an empty placeholder in the following line:")
-                        print(v)
+                        hedy_content_logger.error(
+                            f"There is an issue due to an empty placeholder in line: {v}")
```
[the remark this answers is not in the document]
Comment (babaMar) [reply hedy_content.py:145]: I understand, wanted to remove the `print` statements

File: logger.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+LOG_FORMAT = '[%(asctime)s] %(filename)s %(funcName)s %(lineno)d <%(levelname)s>: %(message)s'
+logging.basicConfig(format=LOG_FORMAT)
+
+app_logger = logging.getLogger('app')
+log_queue_logger = logging.getLogger('querylog')
+jsonbin_logger = logging.getLogger('jsonbin')
+hedy_content_logger = logging.getLogger('hedy_content')
+
+for _logger in (app_logger, log_queue_logger, jsonbin_logger, hedy_content_logger):
+    _logger.setLevel(logging.DEBUG)
```
[the remark this answers is not in the document]
Comment (babaMar) [reply logger.py:12]: This has been essential in tracking the bug

**Comment by babaMar** [comment]
> It is very hard for me to review this PR as there are many changes going on. Please refrain from making changes unrelated to the related issue when submitting a PR. It is still unclear to me what the actual fix is for the related issue.

I completely understand your point. I can also rework the PR disabling the pre-commit hook

File: logger.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (TiBiBa) [reply logger.py:12]: I understand. But please create and issue related to this code, something like "Suggestion to add logger to improve bug tracking" and add this in another PR. This keeps reviewing easy.

**Comment by TiBiBa** [comment]
> > It is very hard for me to review this PR as there are many changes going on. Please refrain from making changes unrelated to the related issue when submitting a PR. It is still unclear to me what the actual fix is for the related issue.
> 
> I completely understand your point. I can also rework the PR disabling the pre-commit hook

All right! Yes I'm not sure what the best approach is here, we just recently starting some styling enforcement using the pre-commit hooks and where afraid before hand that it would alter a lot of code within new pr's. 

@Felienne would it be a good idea to enforce the styling all at once in one PR? Otherwise we will have to keep handling these issues for the next x PRs until all files have passed the new styling.

File: logger.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (babaMar) [reply logger.py:12]: All right, shall I take this out?

**Comment by babaMar** [comment]
@TiBiBa opening a new PR only with the fix.

**Pull request closed without merging**